src/CAD/Vec3d.hh

```diff
@@ -1,86 +1,73 @@
 #pragma once
-#include <iostream>
 #include <cmath>
-
-using namespace std;
-
+#include <iostream>
 
 // create constructor to create the vec_3d objects
-class Vec3D{
-public:
-double x;
-double y;
-double z;
-Vec3D(double x = 0, double y = 0, double z = 0) : x(x), y(y), z(z){}
+class Vec3D {
+ public:
+  double x;
+  double y;
+  double z;
+  Vec3D(double x = 0, double y = 0, double z = 0) : x(x), y(y), z(z) {}
 
-// overload operators for vector math
-friend Vec3D operator+ (const Vec3D& a, const Vec3D& b){
-  return Vec3D(a.x + b.x, a.y + b.y, a.z + b.z);
-}
-friend Vec3D operator- (const Vec3D& a, const Vec3D& b){
-  return Vec3D(a.x - b.x, a.y - b.y, a.z - b.z);
-}
-friend Vec3D operator* (const Vec3D& a, double s){
-  return Vec3D(a.x * s, a.y * s, a.z * s);
-}
-friend Vec3D operator/ (const Vec3D& a, double s){
-  return Vec3D(a.x / s, a.y / s, a.z / s);
-}
+  // overload operators for vector math
+  friend Vec3D operator+(const Vec3D& a, const Vec3D& b) {
+    return Vec3D(a.x + b.x, a.y + b.y, a.z + b.z);
+  }
+  friend Vec3D operator-(const Vec3D& a, const Vec3D& b) {
+    return Vec3D(a.x - b.x, a.y - b.y, a.z - b.z);
+  }
+  friend Vec3D operator*(const Vec3D& a, double s) {
+    return Vec3D(a.x * s, a.y * s, a.z * s);
+  }
+  friend Vec3D operator/(const Vec3D& a, double s) {
+    return Vec3D(a.x / s, a.y / s, a.z / s);
+  }
 
+  Vec3D operator-() const { return Vec3D(-x, -y, -z); }
+  // create three methods to compute various vector properties
+  double mag() const { return sqrt(magsq()); }
+  double magsq() const { return x * x + y * y + z * z; }
+  Vec3D normalize(const Vec3D& a) {
+    if (a.mag() == 0) {
+      return Vec3D(0, 0, 0);
+    } else {
+      return a / a.mag();
+    }
+  }
 
-Vec3D operator- () const {
-  return Vec3D(-x, -y, -z);
-}
-// create three methods to compute various vector properties
-double mag() const {
-  return sqrt(magsq());
-}
-double magsq() const {
-  return x*x+y*y+z*z;
-}
-Vec3D normalize(const Vec3D& a){
-  if (a.mag() == 0){
-    return Vec3D(0,0,0);
+  // compute the dot product as a function
+  friend double dot(const Vec3D& a, const Vec3D& b) {
```

Comment:
All functions external to this had to be pulled in to the class as friend functions. Generally you want your variables to be private and your functions to either be friend functions or class functions anyway, but this was a bit of an extreme case as trying to put any function outside of this class definition looked something like:

```Vec3d.hh:line#, multiple definition of <function>, first defined here: Vec3d.hh:line#```

This seems to imply that somewhere, something is getting included or compiled improperly, but I couldn't find it when I went looking.